clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -626,79 +687,94 @@ static Expected<StringRef> runSYCLPostLink(ArrayRef<StringRef> InputFiles,
     CmdArgs.push_back(File);
   if (Error Err = executeCommands(*SYCLPostLinkPath, CmdArgs))
     return std::move(Err);
-  return *TempFileOrErr;
-}
 
-// This table is used to manage the output table populated by sycl-post-link.
-struct Table {
-  struct SYCLTableEntry {
-    std::string IRFile;
-    std::string PropFile;
-    std::string SymFile;
-  };
+  if (DryRun) {
+    auto ImageFileOrErr = createOutputFile(
+        sys::path::filename(ExecutableName) + ".sycl.split.image", "bc");
+    if (!ImageFileOrErr)
+      return ImageFileOrErr.takeError();
 
-  SmallVector<SYCLTableEntry, 16> Entries;
+    std::vector Modules = {module_split::SplitModule(
+        *ImageFileOrErr, util::PropertySetRegistry(), "")};
+    return Modules;
+  }
+
+  return parseSplitModulesFromFile(*TempFileOrErr);
+}
 
-  SmallVector<std::string, 16> getListOfIRFiles(void) {
-    SmallVector<std::string, 16> Files;
-    for (auto &Entry : Entries) {
-      Files.push_back(Entry.IRFile);
+Error printSplitModules(const std::vector<module_split::SplitModule> &Modules) {
+  for (size_t I = 0, E = Modules.size(); I != E; ++I) {
+    SMDiagnostic Err;
+    LLVMContext C;
+    std::unique_ptr<Module> M = parseIRFile(Modules[I].ModuleFilePath, Err, C);
+    if (!M) {
+      std::string Msg;
+      raw_string_ostream S(Msg);
+      Err.print(ExecutableName.begin(), S);
+      return createStringError(inconvertibleErrorCode(), Msg);
     }
-    return Files;
+
+    errs() << formatv("Split Module. Index: {0}\n", I);
+    M->dump();
   }
 
-  Expected<StringRef> writeSYCLTableToFile(void) {
-    // Create a new file.
-    auto TempFileOrErr =
-        createOutputFile(sys::path::filename(ExecutableName), "table");
-    if (!TempFileOrErr)
-      return TempFileOrErr.takeError();
-    std::error_code EC;
-    raw_fd_ostream TableFile(*TempFileOrErr, EC, sys::fs::OF_None);
-    if (EC)
-      reportError(errorCodeToError(EC));
-    TableFile << "[Code|Properties|Symbols]\n";
-    for (auto &Entry : Entries) {
-      TableFile << Entry.IRFile << "|";
-      TableFile << Entry.PropFile << "|";
-      TableFile << Entry.SymFile << "\n";
-    }
-    return *TempFileOrErr;
+  return Error::success();
+}
+
+static Expected<std::vector<module_split::SplitModule>>
```

Comment:
Please add a comment describing this function (similar to runSYCLPostLinkTool).

Reply:
Added.